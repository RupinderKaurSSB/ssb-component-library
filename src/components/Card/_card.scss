.ssb-card {

  .clickable {
    @include focus-marker;
    @include reset-button;

    img {
      height: 214px;
      width: 380px;
    }

    &.left-orientation {
      display: flex;
      flex-direction: row;

      .card-content {
        height: 214px;
      }
    }

    &.top-orientation {
      display: flex;
      flex-direction: column;
    }

    &:hover {

      .card-content {
        border: 1px solid $ssb-green-3;

        &:not(.with-image) {
          border-top: 5px solid $ssb-green-3;
        }
      }
    }
  }

  .card-content {
    background: $ssb-white;
    border: 1px solid $ssb-dark-2;
    border-top: 5px solid $ssb-green-3;
    box-sizing: border-box;
    cursor: pointer;
    display: inline-block;
    flex-direction: column;
    max-width: 380px;
    min-width: 250px;
    padding: 20px;
    width: 100%;

    &.centered {
      align-items: center;
      display: inline-flex;
      text-align: center;
    }

    &.with-image {
      border-top: 1px solid $ssb-dark-2;
    }
  }

  .download-section {
    @include open-sans;
    align-items: center;
    border: 1px solid $ssb-dark-2;
    cursor: pointer;
    display: flex;
    height: 60px;
    max-width: 380px;
    text-decoration: none;

    &:focus {
      @include focus-ring;
    }

    &:hover {
      border: 1px solid $ssb-green-3;
    }

    .download-icon {
      color: $ssb-green-4;
      font-size: 20px;
      margin: 0 18px;
    }

    span {
      color: $ssb-dark-6;
      font-size: 16px;
    }
  }
}
