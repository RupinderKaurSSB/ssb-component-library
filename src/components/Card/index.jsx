import React from 'react';
import PropTypes from 'prop-types';
import { ArrowRight, Download } from 'react-feather';

const Card = ({
	centered, children, downloadText, fileLocation, href, hrefText, image, imagePlacement, subTitle, title,
}) => (
	<div className="ssb-card">
		<a
			href={href}
			className={`clickable ${imagePlacement === 'left' ? 'left-orientation' : 'top-orientation'}`}
		>
			{image && <div className="card-image">{image}</div>}
			<div className={`card-content${centered ? ' centered' : ''}${image ? ' with-image' : ''}`}>
				{subTitle && <div className="card-subtitle">{subTitle}</div>}
				{title && <div className="card-title">{title}</div>}
				{ children }
				{(!image && !hrefText) && (
					<ArrowRight className="arrow-icon" size={22} />
				)}
				{(!image && hrefText) && (
					<div className="card-action">
						<ArrowRight className="arrow-icon" size={16} />
						<div className="href-text">{hrefText}</div>
					</div>
				)}
			</div>
		</a>
		{fileLocation && (
			<a download href={fileLocation} className="download-section">
				<Download className="download-icon" />
				<span>{downloadText}</span>
			</a>
		)}
	</div>
);

Card.defaultProps = {
	centered: false,
	downloadText: 'Last ned',
	imagePlacement: 'top',
};

Card.propTypes = {
	centered: PropTypes.bool,
	children: PropTypes.node.isRequired,
	downloadText: PropTypes.string,
	fileLocation: PropTypes.string,
	href: PropTypes.string,
	hrefText: PropTypes.string,
	image: PropTypes.element,
	imagePlacement: PropTypes.oneOf(['left', 'top']),
	subTitle: PropTypes.string,
	title: PropTypes.string,
};

export default Card;
