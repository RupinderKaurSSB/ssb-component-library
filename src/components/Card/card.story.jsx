import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import { Globe } from 'react-feather';
import Card from './index';
import Text from '../Text';
import Title from '../Title';
import testImage from '../../media/16x9.png';

storiesOf('Card', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ width: '800px', display: 'flex', justifyContent: 'center' }}>
			<div style={{ marginRight: '10px', display: 'flex' }}>
				<Card title="Tittel" href=" ">
					<Text>Explain something about something with something clever.</Text>
				</Card>
			</div>
			<br />
			<Card title="Tittel" href=" " hrefText="Handling">
				<Text>Explain something about something with something clever.</Text>
			</Card>
		</div>
	))
	.add('With images left', () => (
		<div>
			<Card
				imagePlacement="left"
				image={<img src={testImage} alt="testImage" />}
				href=" "
				subTitle="Artikkel / 11. mai 2019"
				title="Tittel"
			>
				<Text>Explain something about something with something clever.</Text>
			</Card>
		</div>
	))
	.add('With images top', () => (
		<div>
			<Card
				imagePlacement="top"
				image={<img src={testImage} alt="testImage" />}
				href=" "
				subTitle="Artikkel / 11. mai 2019"
				title="Tittel"
			>
				<Text>Explain something about something with something clever.</Text>
			</Card>
		</div>
	))
	.add('Three boxes', () => (
		<div style={{ display: 'grid', gridTemplateColumns: 'repeat(auto, 380px)', gridRowGap: '20px' }}>
			<Card href=" ">
				<Title size={2}>Look at this</Title>
				<Text>Explain something about something with something clever.</Text>
			</Card>
			<Card centered href=" ">
				<Globe size={120} style={{ marginBottom: '20px' }} />
				<Title size={2}>Look at this</Title>
				<Text>This box has the <code>centered</code> prop set to true</Text>
			</Card>
			<Card href=" " fileLocation="./not_a_file.md" downloadText="Download">
				<Title size={2}>Look at this</Title>
				<Text>Explain something about something with something clever.</Text>
			</Card>
		</div>
	));
