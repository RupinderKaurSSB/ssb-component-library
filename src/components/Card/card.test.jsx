import React from 'react';
import {shallow} from 'enzyme';
import Card from './index';

describe('Card component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Card>ProfileBox</Card>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<Card centered>ProfileBox</Card>);
		expect(wrapper.find('.card-content').hasClass('centered')).toEqual(true);
	});

});
