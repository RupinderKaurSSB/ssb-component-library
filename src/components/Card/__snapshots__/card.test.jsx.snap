// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Matches the snapshot 1`] = `
<div
  className="ssb-card"
>
  <a
    className="clickable top-orientation"
  >
    <div
      className="card-content"
    >
      ProfileBox
      <ArrowRight
        className="arrow-icon"
        color="currentColor"
        size={22}
      />
    </div>
  </a>
</div>
`;
